javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+import scala.collection.mutable.ArrayBuffer
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    inner = ArrayBuffer[E](array: _*)
+  }
+
+  private var inner: ArrayBuffer[E] = new ArrayBuffer[E](length)
+
+  final def length(): Int = inner.size
+
+  final def get(i: Int): E = inner(i)
+
+  final def set(i: Int, newValue: E): Unit =
+    inner.insert(i, newValue)
+
+  final def lazySet(i: Int, newValue: E): Unit =
+    set(i, newValue)
+
+  final def getAndSet(i: Int, newValue: E): E = {
+    val ret = get(i)
+    set(i, newValue)
+    ret
+  }
+
+  final def compareAndSet(i: Int, expect: E, update: E): Boolean = {
+    if (get(i) ne expect) false else {
+      set(i, update)
+      true
+    }
+  }
+
+  final def weakCompareAndSet(i: Int, expect: E, update: E): Boolean =
+    compareAndSet(i, expect, update)
+
+  override def toString(): String =
+    inner.mkString("[",", ","]")
```

Comment:
Missing spaces after the two `,` that are Scala code. So basically `("[", ", ", "]")`